app/lib/features/pins/pages/create_pin.dart

```diff
@@ -0,0 +1,359 @@
+            child: pinState.pinDescriptionParams!.htmlBodyDescription.isNotEmpty
+                ? RenderHtml(
+                    text: pinState.pinDescriptionParams!.htmlBodyDescription,
+                    defaultTextStyle: Theme.of(context).textTheme.labelLarge,
+                  )
+                : Text(
+                    pinState.pinDescriptionParams!.plainDescription,
+                    style: Theme.of(context).textTheme.labelLarge,
+                  ),
+          ),
+        ),
+        const SizedBox(height: 20),
+      ],
+    );
+  }
+
+  Widget _buildCreateButton() {
+    return ActerPrimaryActionButton(
+      key: CreatePin.submitBtn,
+      onPressed: _createPin,
+      child: Text(L10n.of(context).create),
+    );
+  }
+
+  Future<void> _createPin() async {
+    if (!_formKey.currentState!.validate()) return;
+    EasyLoading.show(status: L10n.of(context).creatingPin);
+    try {
+      final spaceId = ref.read(selectedSpaceIdProvider);
+      final space = await ref.read(spaceProvider(spaceId!).future);
+      final pinDraft = space.pinDraft();
+      final pinState = ref.read(createPinStateProvider);
+
+      // Pin Title
+      if (pinState.pinTitle != null && pinState.pinTitle!.isNotEmpty) {
+        pinDraft.title(pinState.pinTitle!);
+      }
+
+      // Pin Description
+      if (pinState.pinDescriptionParams != null) {
+        if (pinState.pinDescriptionParams!.htmlBodyDescription.isNotEmpty) {
+          pinDraft.contentHtml(
+            pinState.pinDescriptionParams!.plainDescription,
+            pinState.pinDescriptionParams!.htmlBodyDescription,
+          );
+        } else {
+          pinDraft.contentMarkdown(
+            pinState.pinDescriptionParams!.plainDescription,
+          );
+        }
+      }
+      final pinId = await pinDraft.send();
+
+      // Add Attachments
+      await addAttachment(pinId, pinState);
+
+      EasyLoading.dismiss();
+      if (!mounted) return;
+      EasyLoading.showToast(L10n.of(context).pinCreatedSuccessfully);
+      context.replaceNamed(
+        Routes.pin.name,
+        pathParameters: {'pinId': pinId.toString()},
+      );
+    } catch (e) {
+      EasyLoading.dismiss();
+      if (!mounted) return;
+      EasyLoading.showError(
+        L10n.of(context).errorCreatingPin(e),
+        duration: const Duration(seconds: 3),
+      );
+    }
+  }
+
+  Future<void> addAttachment(EventId pinId, CreatePinState pinState) async {
+    final acterPin = await ref.read(pinProvider(pinId.toString()).future);
+    final manager = await acterPin.attachments();
+    if (!mounted) return;
+    for (final attachment in pinState.pinAttachmentList) {
+      await handleAttachmentSelected(
+        context: context,
+        ref: ref,
+        manager: manager,
+        attachments: attachment.path != null ? [File(attachment.path!)] : [],
+        title: attachment.title,
+        link: attachment.link,
+        attachmentType: attachment.attachmentType,
+      );
+    }
+  }
```

Comment:
This is the for loop which use to add all attachments with existing `handleAttachmentSelected`.